usaspending_api/etl/award_helpers.py

```diff
@@ -145,11 +129,41 @@ def update_awards(award_tuple: Optional[tuple] = None) -> int:
         "  WHERE e.award_id = a.id "
     )
 
-    sql_prune = prune_empty_awards_cte
-
     sql_update = _sql_update.format(earliest_transaction_cte, latest_transaction_cte, aggregate_transaction_cte)
     # We don't need to worry about this double counting awards, because if it's deleted in the first step it can't be updated!
-    return execute_database_statement(sql_prune, [award_tuple]) + execute_database_statement(sql_update, values)
+    return prune_empty_awards(award_tuple) + execute_database_statement(sql_update, values)
+
+
+def prune_empty_awards(award_tuple: Optional[tuple] = None) -> int:
+    _find_empty_awards_sql = str(
+        "SELECT a.id "
+        "FROM awards a "
+        "LEFT JOIN transaction_normalized t "
+        "ON t.award_id = a.id "
+        "WHERE t is null {}"
```

Comment:
Adding ```WHERE t is null``` was a good move, I think it will prevent any odd behavior when all of an award's transactions were deleted then re-added within the same load.

Reply:
I would go further and say that these operations should be context-independent moving forward. update_awards() could be called at the end of any other process, perhaps one that was only taking away transactions or just modifying them. As I best understand, the above query defines what a dead award is, not just what awards a particular loader might be interested in removing.

Reply:
> As I best understand, the above query defines what a dead award is, not just what awards a particular loader might be interested in removing.

It is working off the assumption that transactions are correctly linked to awards using the FK, not examining the actual award data in transactions. This should most likely only be used in context of transaction loads.

Reply:
Awards are a dependent table on transactions, and use that dependency as a source of truth. If the foreign keys in transactions are wrong, then I would argue that the awards table SHOULD accurately reflect that falsehood, just like if broker were to pass us inaccurate data inside some transaction field, we would pass that incorrect value on to the website. 

This probably isn't a useful conversation to have here, however.